doc/build-unix.md

```diff
@@ -164,15 +163,13 @@ turned off by default.  See the configure options for upnp behavior desired:
 
 Berkeley DB
 -----------
-It is recommended to use Berkeley DB 4.8. If you have to build it yourself,
-you can use [the installation script included in contrib/](/contrib/install_db4.sh)
-like so:
-
-```shell
-./contrib/install_db4.sh `pwd`
-```
 
-from the root of the repository.
+BerkeleyDB 4.7 through 5.3 (inclusive) can be used. Most Linux distributions ship BerkeleyDB 5.3.
+The `wallet.dat` files will be compatible as the database file format does not change within these versions.
+The database transaction log files do have incompatible formats, but these log files are only relevant if the wallet closes uncleanly.
+If a wallet is cleanly closed, the log files will be removed. If it is not cleanly closed, then a
+build using the same version of BerkeleyDB will need to be used to open and close that wallet to guarantee that
+no data is lost.
```

Comment:
Part after "will need to be used to to guarantee that no data is lost" seems misleading and not specific enough when there's no good indication of lost data. Would change to "If not cleanly closed, then builds with the newer versions of BerkeleyDB will recover using all available log data, but builds with older versions of BerkeleyDB will open the wallet potentially not using all of the recovery data."

re: https://github.com/bitcoin/bitcoin/pull/18870#discussion_r421604853

> > Or is the wallet allowed to load with incomplete data?
> 
> The wallet is allowed to load. The logs are moved to a backup directory and an error is logged to both the debug.log and db.log. Unless the user is actively monitoring one of those files, they wouldn't know that there is an error and potential data loss.

Ouch. With or without this PR, it seems like dangerous to just throw away logs from future versions like that. I think separately we should implement a fix to prevent this type of data loss, even it is something really cheesy like writing a `wallet.db.ver` file each time we load a wallet with the current Berkeley DB version number, and refusing to load any wallet when (1) `wallet.db.ver` is present and (2) leftover `database/log.??????????` files are present and (3) `wallet.db.ver` is greater than the current berkeley db version. Error message for refusing to load could be something like "Wallet [name] not shut down cleanly, and was last loaded using a newer version of Berkeley DB. Wallet software built with Berkeley DB version %s or later is required to recover this wallet."

All of this makes me less enthusiastic than other reviewers are about this PR. Simplifying the build a little vs creating a new scenario for data loss does not seem like an appealing tradeoff to me, especially when we may want to rely on logs more in the future to deal with wallet performance problems like #18886 / #18904. But it's not a crazy tradeoff and people do love simple build steps.

Reply:
I think we could just remove the behavior where we remove the log files and retry, especially if we are getting a `DB_RUNRECOVERY` error, which, as the name implies, means that BDB thinks we should be doing recovery procedures. This is the error that it returns when the wrong log version is found.

Reply:
> I think we could just remove the behavior where we remove the log files and retry, especially if we are getting a `DB_RUNRECOVERY` error, which, as the name implies, means that BDB thinks we should be doing recovery procedures. This is the error that it returns when the wrong log version is found.

Yes, I might have been too pessimistic about how often the recovery case would happen. That idea seems simpler and better than my suggestion if it works and doesn't cause headaches for users needing to do manual recoveries. If it does cause headaches, something like the  `wallet.db.ver ` idea could always be added later to restore automatic recovery when there isn't a risk from version mismatch